ironfish/src/consensus/verifier.test.ts

```diff
@@ -373,6 +373,166 @@ describe('Verifier', () => {
     })
   })
 
+  describe('Before postive block mine time activation', () => {
+    const nodeTest = createNodeTest()
+    let block: Block
+    let header: BlockHeader
+    let prevHeader: BlockHeader
+
+    beforeAll(async () => {
+      const { chain } = await nodeTest.createSetup()
+      block = await useMinerBlockFixture(
+        chain,
+        chain.consensus.parameters.disallowNegativeBlockMineTime - 1,
+      )
+      header = block.header
+      const previousBlock = block.header.previousBlockHash
+      const previousHeader = await chain.getHeader(previousBlock)
+      Assert.isNotNull(previousHeader)
+      prevHeader = previousHeader
+    })
+
+    it('fails validation when timestamp is too low', async () => {
+      jest
+        .spyOn(global.Date, 'now')
+        .mockImplementationOnce(() => prevHeader.timestamp.getTime() + 1 * 1000)
+      header.timestamp = new Date(
+        prevHeader.timestamp.getTime() -
+          (nodeTest.chain.consensus.parameters.allowedBlockFutureSeconds + 2) * 1000,
+      )
+
+      expect(await nodeTest.verifier.verifyBlockAdd(block, prevHeader)).toMatchObject({
+        reason: VerificationResultReason.BLOCK_TOO_OLD,
+        valid: false,
+      })
+    })
+
+    it('fails validation when timestamp is too far in the future', async () => {
```

Comment:
It looks like there's tests in here already for TOO_FAR_IN_FUTURE: https://github.com/iron-fish/ironfish/blob/master/ironfish/src/consensus/verifier.test.ts#L397

Since you aren't modifying that behavior and we already have tests there, I don't think we should duplicate the tests

Reply:
Yeah I saw the existing test for TOO_FAR_IN_FUTURE case. I added them here to make sure the block sequence before and after activation are behaved as expected for TOO_FAR_IN_FUTURE case. The existing test does not test the activation flag params.